src/flakeModule.nix

```diff
@@ -0,0 +1,138 @@
+    (builtins)
+    all
+    concatMap
+    head
+    isList
+    length
+    mapAttrs
+    tail
+    zipAttrsWith
+    ;
+  inherit
+    (lib)
+    getValues
+    mkOption
+    mkOptionType
+    showDefs
+    showOption
+    types
+    ;
+
+  # ensures we always end up with a list of lists
+  # if more than one definition
+  mergeHarvesterOption = loc: defs: let
+    list = getValues defs;
+  in
+    if length list == 1
+    then head list
+    else if all isList list
+    then
+      concatMap (l:
+        if isList l
+        then l
+        else [l])
+      list
+    else throw "Cannot merge definitions of `${showOption loc}'. Definition values:${showDefs defs}";
+
+  harvesterType = mkOptionType {
+    name = "harvest path";
+    description = "defines the path as list (or list of lists) from where to harvest/winnow/pick targets into the Nix-CLI compatible output form";
+    descriptionClass = "noun";
+    merge = mergeHarvesterOption;
+    emptyValue = {value = [];};
+    inherit
+      (types.either
+        (types.listOf types.nonEmptyStr)
+        (types.listOf (types.listOf types.nonEmptyStr)))
+      check
+      ;
+  };
+
+  cellBlocksType = mkOptionType {
+    name = "cell block";
+    description = "defines the global set of output/target types of the project";
+    descriptionClass = "noun";
+    inherit (validators.CellBlocks) check;
+  };
+
+  opt = options.std;
+  cfg = config.std;
+in {
+  _file = ./flakeModule.nix;
+  options = {
+    std = {
+      grow = mkOption {
+        description = "Orderly 'grow' a project from Standard Cells & Cell Blocks";
+        type = types.submodule {
+          options = {
+            cellsFrom = mkOption {
+              description = "Where Standard discovers Cells from.";
+              type = types.path;
+            };
+            cellBlocks = mkOption {
+              description = "Declaration of all Cell Blocks used in the project.";
+              type = cellBlocksType;
+            };
+            nixpkgsConfig = mkOption {
+              description = "Nixpkgs configuration applied to `inputs.nixpkgs` (if that input exists).";
+              type = types.attrs;
+            };
+          };
+        };
+      };
+      pick = mkOption {
+        description = "Pick Standard outputs. Like 'harvest' but remove the system for outputs that are system agnostic.";
+        type = types.attrsOf harvesterType;
+      };
+      winnowIf = mkOption {
+        description = "predicate for winnow";
+        type = types.attrsOf (types.functionTo (types.functionTo types.bool));
+      };
+      winnow = mkOption {
+        description = "Winnow Standard outputs. Like 'harvest', but with a filter predicate 'winnowIf'.";
+        type = types.attrsOf harvesterType;
+      };
+      harvest = mkOption {
+        description = "Harvest Standard outputs into a Nix-CLI-compatible form (a.k.a. the 'official' flake schema).";
+        type = types.attrsOf harvesterType;
+      };
+    };
+  };
```

Comment:
I'd recommend to add examples to all of the options, and perhaps expand the documentation a bit, possibly by linking to existing documentation. `description` is markdown, ahead of the Nixpkgs/NixOS switch to markdown.

Reply:
An complete example in the form of a flake template would also be helpful.

Reply:
> An complete example in the form of a flake template would also be helpful.

That's in planning as part of https://github.com/divnix/std/issues/211

I don't want to treat it in an ad-hoc way...